cmd/config-doc-gen/main.go

```diff
@@ -0,0 +1,287 @@
+	var fieldDocs []FieldDoc
+	fieldType := exprToString(fset, field.Type)
+	fieldTag := extractTag(field.Tag)
+
+	for _, name := range field.Names {
+		fieldDoc := FieldDoc{
+			Name:    name.Name,
+			Type:    fieldType,
+			Tag:     fieldTag,
+			Comment: extractComment(field.Doc),
+		}
+		fieldDocs = append(fieldDocs, fieldDoc)
+	}
+
+	// Handle nested struct
+	if ident, ok := field.Type.(*ast.Ident); ok && ident.Obj != nil {
+		if typeSpec, ok := ident.Obj.Decl.(*ast.TypeSpec); ok {
+			if structType, isStruct := typeSpec.Type.(*ast.StructType); isStruct {
+				nestedStructDoc := StructDoc{
+					Name:    ident.Name,
+					Comment: extractComment(typeSpec.Doc),
+				}
+				for _, nestedField := range structType.Fields.List {
+					nestedStructDoc.Fields = append(nestedStructDoc.Fields, processField(fset, nestedField, structs)...)
+				}
+				structs[ident.Name] = nestedStructDoc
+			}
+		}
+	}
+
+	return fieldDocs
+}
+
+// parseFile parses a Go source file and extracts struct information.
+func parseAST(node *ast.File) []StructDoc {
+	var structs []StructDoc
+	structMap := make(map[string]StructDoc)
+	defaultValues := make(map[string]map[string]interface{}) // Store default values for structs
+
+	// Create a new token file set
+	fset := token.NewFileSet()
+
+	// Traverse the AST to find struct type declarations and functions
+	ast.Inspect(node, func(n ast.Node) bool {
+		switch n := n.(type) {
+		case *ast.GenDecl: // Handle struct declarations
+			if n.Tok != token.TYPE {
+				return true
+			}
+			for _, spec := range n.Specs {
+				typeSpec, ok := spec.(*ast.TypeSpec)
+				if !ok {
+					continue
+				}
+				structType, ok := typeSpec.Type.(*ast.StructType)
+				if !ok {
+					continue
+				}
+
+				structDoc := StructDoc{
+					Name:    typeSpec.Name.Name,
+					Comment: extractComment(n.Doc),
+				}
+
+				for _, field := range structType.Fields.List {
+					structDoc.Fields = append(structDoc.Fields, processField(fset, field, structMap)...)
+				}
+
+				structMap[structDoc.Name] = structDoc
+			}
+
+		case *ast.FuncDecl: // Handle `createDefaultConfig()`
+			if n.Name.Name == "createDefaultConfig" {
+				structName, defaults := extractDefaultValues(fset, n)
+
+				if structName != "" {
+					defaultValues[structName] = defaults
+				}
+			}
+		}
+		return true
+	})
+
+	// Apply extracted defaults to structs
+	for structName, defaults := range defaultValues {
+		if structDoc, exists := structMap[structName]; exists {
+			for i, field := range structDoc.Fields {
+				if val, found := defaults[field.Name]; found {
+					structDoc.Fields[i].DefaultValue = val
+				}
+			}
+			structMap[structName] = structDoc
+		}
+	}
+
+	// Convert map to slice
+	for _, structDoc := range structMap {
+		structs = append(structs, structDoc)
+	}
+	return structs
+}
+
+// function to extract default value from struct
+func extractDefaultValues(fset *token.FileSet, fn *ast.FuncDecl) (string, map[string]interface{}) {
```

Comment:
we will need unit tests to ensure that this logic does not miss fields (like embedded structs).